mypy.ini

```diff
@@ -326,9 +326,6 @@ disable_error_code = type-arg,var-annotated
 [mypy-plasmapy.particles.atomic]
 disable_error_code = arg-type,assignment,misc,no-any-return,no-untyped-call,no-untyped-def,union-attr
 
-[mypy-plasmapy.particles.decorators]
-disable_error_code = arg-type,assignment,call-arg,index,misc,no-any-return,no-untyped-call,no-untyped-def,operator,return-value,type-arg,union-attr,var-annotated
```

Comment:
These are the mypy errors that existed in `plasmapy.particles.decorators` before this pull request. My process was essentially to:

1. Remove one error code and find the errors that mypy was reporting
2. Either fix those errors, or add a `#type: ignore` comment if the reported error wasn't actually a problem with the code
3. `GOTO 1`